plugins/linux-pipewire/pipewire.c

```diff
@@ -765,14 +765,24 @@ static void process_video_sync(obs_pipewire_stream *obs_pw_stream)
 		}
 
 #if PW_CHECK_VERSION(1, 2, 0)
+		if (obs_pw_stream->sync.acquire_syncobj_fd != -1) {
+			close(obs_pw_stream->sync.acquire_syncobj_fd);
+			obs_pw_stream->sync.acquire_syncobj_fd = -1;
+		}
+		if (obs_pw_stream->sync.release_syncobj_fd != -1) {
+			close(obs_pw_stream->sync.release_syncobj_fd);
+			obs_pw_stream->sync.release_syncobj_fd = -1;
+		}
+
 		if (synctimeline && (buffer->n_datas == (planes + 2))) {
 			assert(buffer->datas[planes].type == SPA_DATA_SyncObj);
 			assert(buffer->datas[planes + 1].type == SPA_DATA_SyncObj);
 
-			obs_pw_stream->sync.acquire_syncobj_fd = buffer->datas[planes].fd;
+			obs_pw_stream->sync.acquire_syncobj_fd = fcntl(buffer->datas[planes].fd, F_DUPFD_CLOEXEC, 5);
```

Comment:
Five?

Reply:
It just requires that the returned descriptor, is an integer greater or equal to five, which is a common pattern.

We already do it for the PipeWire core file descriptor.